### ISSUE TITLE
Crash on animation heavy site

### ISSUE THREAD

**Issue by treba123** [servo/webrender#2953]
Visiting https://ncase.me/trust/ reliably crashes Firefox for me when webrender is enabled. Tested on [firefox nightly flatpak version](https://firefox-flatpak.mojefedora.cz/) on fedora 28, gnome on wayland

The console output:
```
Attempting load of libEGL.so
WebGL(0x7f393267a800)::ForceLoseContext
JavaScript warning: https://ncase.me/trust/js/lib/pixi.min.js?v11, line 14: Error: WebGL warning: drawElements: This operation requires zeroing texture data. This is slow.
[Child 228, Chrome_ChildThread] WARNING: pipe error (37): Connection reset by peer: file /run/build/firefox/ipc/chromium/src/chrome/common/ipc_channel_posix.cc, line 353
[Child 228, Chrome_ChildThread] WARNING: pipe error (3): Connection reset by peer: file /run/build/firefox/ipc/chromium/src/chrome/common/ipc_channel_posix.cc, line 353
[Child 187, Chrome_ChildThread] WARNING: pipe error (3): Connection reset by peer: file /run/build/firefox/ipc/chromium/src/chrome/common/ipc_channel_posix.cc, line 353
/app/bin/firefox: line 3:     3 Segmentation fault      (core dumped) firefox-bin $*
```

**Comment by kvark** [comment]
FTR, I'm on "63.0a1 (2018-08-07) (64-bit)" /Linux and it doesn't crash.

**Comment by Darkspirit** [comment]
This might be https://github.com/xhorak/firefox-devedition-flatpak/issues/42 ?

**Comment by treba123** [comment]
Thanks for the reply! I'll check with a non-flatpak nightly and in a non-wayland session to make sure it's not a bug in webrender.

**Comment by stransky** [comment]
I tested latest nightly on Wayland, webrender enabled, layers.acceleration.force-enabled=true, fedora 28/gnome-shell and it does not crash although there are some warning lines:

---------------------------------------------------------------------------------------------------------------------------
Attempting load of libEGL.so
libEGL warning: MESA-LOADER: failed to retrieve device information

[Child 32565, Main Thread] WARNING: Failed to find a compatible config.: file /home/komat/tmp676-trunk-gtk3/src-wayland/gfx/gl/GLContextProviderEGL.cpp, line 1085
JavaScript warning: https://ncase.me/trust/js/lib/pixi.min.js?v11, line 16: Error: WebGL warning: Failed to create WebGL context: WebGL creation failed: 
* tryNativeGL
* Exhausted GL driver options.
[Child 32565, Main Thread] WARNING: Failed to find a compatible config.: file /home/komat/tmp676-trunk-gtk3/src-wayland/gfx/gl/GLContextProviderEGL.cpp, line 1085
JavaScript warning: https://ncase.me/trust/js/lib/pixi.min.js?v11, line 16: Error: WebGL warning: Failed to create WebGL context: WebGL creation failed: 
* tryNativeGL
* Exhausted GL driver options.
---------------------------------------------------------------------------------------------------------------------------

That may be caused by wrong GL visual config or some issues with Wayland/EGL. It would be great to have a backtrace of the crash. Also which GFX card do you have?

**Comment by treba123** [comment]
I'll try to make a backtrace as soon as I have time. It's a intel haswell gpu, mesa from flatpak is at 18.0.4

**Comment by stransky** [comment]
I can reproduce the crash on debug build. It's at SurfaceFormatToImageFormat(). It's called from TextureHost.cpp and gfx::SurfaceFormat::R8G8B8X8 which is unsupported by WebRender.

**Comment by stransky** [comment]
Filed as https://bugzilla.mozilla.org/show_bug.cgi?id=1482350

**Comment by treba123** [comment]
Fixed via #3337 and https://bugzilla.mozilla.org/show_bug.cgi?id=1482350
